### PR TITLE
Cleanup: StringBuilder issues

### PR DESCRIPTION
**Pull request by XenoAmess**
## Related issues

- Fixes #

## Ready?

- [ ] Added unit tests for fixed bug/feature
- [x] Passing all unit tests
- [x] Complete build `./mvnw clean verify` passes (checked automatically by travis)
- [ ] Added (in-code) documentation (if needed)

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">This changeset introduces 0 new violations, 0 new errors and 0 new configuration errors,<br />
removes 0 violations, 0 errors and 0 configuration errors.<br />
<a href="https://chunk.io/pmd/f6c00798ac124e1abb896c686e490e87/diff/index.html">Full report</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: pmd-core/src/main/java/net/sourceforge/pmd/renderers/TextColorRenderer.java [hunk 1]
```diff
@@ -125,17 +125,12 @@ public void end() throws IOException {
...
             }
-            buf.append(
-                    this.green + "    src:  " + this.cyan + lastFile.substring(lastFile.lastIndexOf(File.separator) + 1)
-                            + this.colorReset + ":" + this.cyan + rv.getBeginLine()
-                            + (rv.getEndLine() == -1 ? "" : ":" + rv.getEndLine()) + this.colorReset + PMD.EOL);
-            buf.append(this.green + "    rule: " + this.colorReset + rv.getRule().getName() + PMD.EOL);
-            buf.append(this.green + "    msg:  " + this.colorReset + rv.getDescription() + PMD.EOL);
-            buf.append(this.green + "    code: " + this.colorReset + this.getLine(lastFile, rv.getBeginLine()) + PMD.EOL
-                    + PMD.EOL);
+            buf.append(this.green).append("    src:  ").append(this.cyan).append(lastFile.substring(lastFile.lastIndexOf(File.separator) + 1)).append(this.colorReset).append(":").append(this.cyan).append(rv.getBeginLine()).append(rv.getEndLine() == -1 ? "" : ":" + rv.getEndLine()).append(this.colorReset).append(PMD.EOL);
```
Comment (oowekyala) [inline pmd-core/src/main/java/net/sourceforge/pmd/renderers/TextColorRenderer.java:130]: This line is way too long

File: pmd-java/src/test/java/net/sourceforge/pmd/lang/java/ast/AccessNodeTest.java [hunk 2]
```diff
@@ -139,9 +139,9 @@ public void testPackagePrivate() {
 
 
     private static String makeAccessJavaCode(String[] access, String declRest) {
-        String result = "public class Test { ";
+        StringBuilder result = new StringBuilder("public class Test { ");
```
Comment (oowekyala) [inline pmd-java/src/test/java/net/sourceforge/pmd/lang/java/ast/AccessNodeTest.java:142]: Please don't edit this file as it has been removed in another branch

File: pmd-vm/src/main/java/net/sourceforge/pmd/lang/vm/ast/TokenMgrError.java [hunk 3]
```diff
@@ -101,7 +101,7 @@ protected static final String addEscapes(final String str) {
                 ch = str.charAt(i);
                 if (ch < 0x20 || ch > 0x7e) {
                     final String s = "0000" + Integer.toString(ch, 16);
-                    retval.append("\\u" + s.substring(s.length() - 4, s.length()));
+                    retval.append("\\u").append(s.substring(s.length() - 4, s.length()));
```
Comment (oowekyala) [inline pmd-vm/src/main/java/net/sourceforge/pmd/lang/vm/ast/TokenMgrError.java:104]: Please don't edit this file as it has been removed in another branch

File: pmd-core/src/main/java/net/sourceforge/pmd/renderers/CodeClimateRenderer.java [hunk 4]
```diff
@@ -145,27 +145,27 @@ private int getRemediationPoints() {
     }
 
     private <T> String getBody() {
-        String result = "## " + rule.getName() + "\\n\\n" + "Since: PMD " + rule.getSince() + "\\n\\n" + "Priority: "
-            + rule.getPriority() + "\\n\\n"
-            + "[Categories](https://github.com/codeclimate/platform/blob/master/spec/analyzers/SPEC.md#categories): "
-            + Arrays.toString(getCategories()).replaceAll("[\\[\\]]", "") + "\\n\\n"
-            + "[Remediation Points](https://github.com/codeclimate/platform/blob/master/spec/analyzers/SPEC.md#remediation-points): "
-            + getRemediationPoints() + "\\n\\n" + cleaned(rule.getDescription());
+        StringBuilder result = new StringBuilder("## " + rule.getName() + "\\n\\n" + "Since: PMD " + rule.getSince() + "\\n\\n" + "Priority: "
```
Comment (oowekyala) [inline pmd-core/src/main/java/net/sourceforge/pmd/renderers/CodeClimateRenderer.java:148]: This stringbuilder uses a concatenation as an argument

**Review by oowekyala — COMMENTED** [review]
Thanks, there are some things I'd like you to fix or revert though

File: pmd-java/src/test/java/net/sourceforge/pmd/lang/java/ast/AccessNodeTest.java (hunk 2 above)
Comment (XenoAmess) [reply pmd-java/src/test/java/net/sourceforge/pmd/lang/java/ast/AccessNodeTest.java]: @oowekyala reverted&rebased

File: pmd-vm/src/main/java/net/sourceforge/pmd/lang/vm/ast/TokenMgrError.java (hunk 3 above)
Comment (XenoAmess) [reply pmd-vm/src/main/java/net/sourceforge/pmd/lang/vm/ast/TokenMgrError.java]: @oowekyala reverted&rebased

File: pmd-core/src/main/java/net/sourceforge/pmd/renderers/TextColorRenderer.java (hunk 1 above)
Comment (XenoAmess) [reply pmd-core/src/main/java/net/sourceforge/pmd/renderers/TextColorRenderer.java]: @oowekyala line splited.

File: pmd-core/src/main/java/net/sourceforge/pmd/renderers/CodeClimateRenderer.java (hunk 4 above)
Comment (XenoAmess) [reply pmd-core/src/main/java/net/sourceforge/pmd/renderers/CodeClimateRenderer.java]: @oowekyala my bad. re-done&rebased.

**Comment by oowekyala** [comment]
Thanks, I'll merge this now. 

Just a tip about github etiquette: don't force-push a branch that has an open PR, prefer to add a fixup commit.

Finally a comment about your entire series of PRs: I think this would have been easier for us if you had opened a single PR with one commit per refactoring. This reduces overhead for the one merging, and prevents merge conflicts between your changes (which happened between #2726 and #2734)

File: pmd-core/src/main/java/net/sourceforge/pmd/cli/PMDCommandLineInterface.java [hunk 5]
```diff
@@ -60,10 +60,10 @@ public static String buildUsageText() {
     public static String buildUsageText(JCommander jcommander) {
         StringBuilder usage = new StringBuilder();
 
-        String allCommandsDescription = null;
+        StringBuilder allCommandsDescription = null;
         if (jcommander != null && jcommander.getCommands() != null) {
             for (String command : jcommander.getCommands().keySet()) {
-                allCommandsDescription += jcommander.getCommandDescription(command) + PMD.EOL;
+                allCommandsDescription.append(jcommander.getCommandDescription(command)).append(PMD.EOL);
```
Comment (tweimer) [inline pmd-core/src/main/java/net/sourceforge/pmd/cli/PMDCommandLineInterface.java:66]: @XenoAmess This introduces a NPE.

Comment (XenoAmess) [reply pmd-core/src/main/java/net/sourceforge/pmd/cli/PMDCommandLineInterface.java:66]: @tweimer 
my bad.
btw: is this funtion covered by any test?

Comment (XenoAmess) [reply pmd-core/src/main/java/net/sourceforge/pmd/cli/PMDCommandLineInterface.java:66]: > @XenoAmess This introduces a NPE.

@tweimer fixed in https://github.com/pmd/pmd/pull/2745.

**Pull request merged**